src/meta/protos/proto/metadata.proto

```diff
@@ -28,6 +28,11 @@ message DataSchema {
 
   // Other optional table info
   map<string, string> metadata = 2;
+
+  uint32 max_column_id = 3;
+
+  // Map column name to column id
+  map<string, uint32> column_id_map = 4;
```

Comment:
How to get a field instance by column-id?

As `self.fields` is a vec of `field`, is column-id the index of a field in `self.fields`?

Reply:
first get field use index, than get column id using field.name in column_id_map.